src/back/space/src/repo/table_space.cpp

```diff
@@ -0,0 +1,228 @@
+{
+	PagingResult<model::Space> data;
+
+	auto trx = _pg->Begin(storages::postgres::Transaction::RO);
+	auto res = trx.Execute(kSelectSpaceAvailable, userId, offset, limit);
+	data.items = res.AsContainer<decltype(data.items)>(pg::kRowTag);
+	res = trx.Execute(kCountSpaceAvailable, userId);
+	data.total = res.AsSingleRow<int64_t>();
+	trx.Commit();
+	return data;
+}
+
+const storages::postgres::Query kSelectByUserId{
+	R"~(
+		SELECT s.id, s.name, s.key, s.requestsAllowed, s.createdAt
+		FROM space s
+		LEFT JOIN space_user su ON s.id = su.spaceId
+		WHERE su.userId = $1
+		OFFSET $2 LIMIT $3
+	)~",
+	storages::postgres::Query::Name{"select_space_by_user_id"},
+};
+
+const storages::postgres::Query kCountByUserId{
+	R"~(
+		SELECT count(*)
+		FROM space s
+		LEFT JOIN space_user su ON s.id = su.spaceId
+		WHERE su.userId = $1
+	)~",
+	storages::postgres::Query::Name{"count_space_by_user_id"},
+};
+
+PagingResult<model::Space> Space::SelectByUserId(const std::string& userId, const int offset, const int limit)
+{
+	PagingResult<model::Space> data;
+
+	auto trx = _pg->Begin(storages::postgres::Transaction::RO);
+	auto res = trx.Execute(kSelectByUserId, userId, offset, limit);
+	data.items = res.AsContainer<decltype(data.items)>(pg::kRowTag);
+	res = trx.Execute(kCountByUserId, userId);
+	data.total = res.AsSingleRow<int64_t>();
+	trx.Commit();
+	return data;
+}
+
+const storages::postgres::Query kSelectSpaceByKey{
+	"SELECT 1 FROM space WHERE key=$1",
+	storages::postgres::Query::Name{"select_space_by_key"},
+};
+
+bool Space::IsExists(const std::string& key) {
+	auto res = _pg->Execute(storages::postgres::ClusterHostType::kMaster, kSelectSpaceByKey, key);
+
+	return !res.IsEmpty();
+}
+
+const storages::postgres::Query kSelectWithDateClauseForOwner {
+	R"~(
+		SELECT 1
+		FROM space s
+		LEFT JOIN space_user su ON s.id = su.spaceId
+		WHERE s.createdAt >= $1 AND su.userId = $2 AND su.isOwner = true
+	)~",
+	storages::postgres::Query::Name{"select_space_with_date_clause_for_owner"},
+};
+
+bool Space::IsReadyForCreationByTime(const std::string& userId) {
+	const auto minuteAgo = std::chrono::system_clock::now() - std::chrono::minutes(1);
+	const auto minuteAgoTimestamp = std::chrono::duration_cast<std::chrono::seconds>(minuteAgo.time_since_epoch()).count();
+
+	auto res = _pg->Execute(storages::postgres::ClusterHostType::kMaster, kSelectWithDateClauseForOwner, minuteAgoTimestamp, userId);
+
+	return res.IsEmpty();
+}
+
+const storages::postgres::Query kCountSpacesWithUser {
+	R"~(
+		SELECT count(*)
+		FROM space s
+		LEFT JOIN space_user su ON s.id = su.spaceId
+		WHERE su.userId = $1
+	)~",
+	storages::postgres::Query::Name{"count_spaces_with_user"},
+};
+
+int Space::GetCountSpacesWithUser(const std::string& userId) {
+	const auto res = _pg->Execute(storages::postgres::ClusterHostType::kMaster, kCountSpacesWithUser, userId);
+
+	int64_t count;
+	if (!res.IsEmpty())
+		count = res.Front()[0].As<int64_t>();
+
+	return count;
+}
```

Comment:
Там где явно возвращается только одна строка, давай использовать AsSingleRow
```suggestion
int64_t Space::GetCountSpacesWithUser(const std::string& userId) {
	const auto res = _pg->Execute(storages::postgres::ClusterHostType::kMaster, kCountSpacesWithUser, userId);
	if (res.IsEmpty())
		return 0;
	return res.AsSingleRow<int64_t>();
}
```

Reply:
Готово (везде).